### PR TITLE
[Enhancement](txn) Block new insert into if schema change happens during transaction

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java [hunk 1]
```diff
@@ -151,6 +151,8 @@ public enum ConnectType {
     protected volatile long currentDbId = -1;
     // Transaction
     protected volatile TransactionEntry txnEntry = null;
+    protected volatile boolean isFirstTxnInsert = false;
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/qe/ConnectContext.java:154]: it's better to move the 2 variables into TransactionEntry

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Review by mymeiyi — APPROVED** [review]

**Review by dataroaring — DISMISSED** [review]
LGTM

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run cloud_p0

**Comment by Yukang-Lian** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**